docs/dev/design/telemetry.rst

```diff
@@ -234,20 +249,21 @@ The final file to be saved would have the following information:
         ]
      },
      "python": "3.7",
-     "os": {
-         "name": "ubuntu",
-         "version": "18.04.5"
-     }
+     "os": "ubuntu-18.04.5"
    }
 
 Storage
 -------
 
+All this information can be collected after the build has finished,
+and we can store it in a dedicated database (telemetry), using Django's models.
+
 Since this information isn't sensitive,
 we should be fine saving this data even if the project/version is deleted.
 As we don't care about historical data,
 we can save the information per-version and from their latest build only.
+And delete old data if it grows too much.
 
-We can collect data for one year,
-export it to cloud storage after being analyzed (maybe share this data publicity),
-and remove it from our database if it takes too much space.
+Should we make heavy use of JSON fields?
+Or try to avoid nesting structures as possible?
+Like config.user/config.final vs user_config/final_config.
```

Comment:
I'm fine using `config.user` and `config.final`. Nesting these allow us to have a better understanding of the structure and I haven't read that this is a problem in postgres. I was able to make queries really easy when using nested fields and they look great to me. So 👍🏼 on using them.

Reply:
Another downside of using nested types is that we lose the definition of these in our models. Like having a plain config attribute that can contain anything vs having a specific attribute :thinking:

Reply:
I'm not sure I'm following here. We decided to use a `JSONField` for multiple reasons, but one of them is that we can change the structure without having to modify the definition of the models. So, there could be config files with `build.jobs` and others without that field at all (not even null --the key won't be present) and that's totally fine.

The model should be as simple as:

```python
class BuildData(models.Model):
  data = models.JSONField()
```

The structure of the JSON may vary over time without requiring modifying the model.

Reply:
The idea I had was to have several fields, and only use json on the fields that have data that changes over time (config file), instead of just one big json field.

Reply:
I understand. Why do you think it's better to have several regular fields and JSON ones just for some of them?

I personally don't see too much value in that approach and I'd like to not deal with database migrations overtime for this database. I'd start with the most simple structure (one big JSON field) and grow from there if we find that just one field is not enough or it does not work as we expected.